src/components/Header.jsx

```diff
@@ -0,0 +1,123 @@
+import { useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import './Header.scss';
+
+const Header = () => {
+  const [select, setSelect] = useState(1);
+  const [widthScreen, setWidthScreen] = useState(window.innerWidth);
+  useEffect(() => {
+    window.addEventListener('resize', () => setWidthScreen(window.innerWidth));
+  }, []);
+
+  const [menuDisplay, setMenuDisplay] = useState(window.innerWidth > 800);
+  useEffect(() => {
+    document.querySelector('.menuContainer').style.display = menuDisplay
+      ? 'flex'
+      : 'none';
+  }, [menuDisplay]);
+
+  const handleMenu = (click) => {
+    if (click === 'checkbox' || click === 'resize') {
+      setMenuDisplay(!menuDisplay);
+    } else if (click === 'link' && widthScreen <= 800) {
+      setMenuDisplay(false);
+    }
+    return null;
+  };
+
+  useEffect(() => {
+    if (widthScreen > 800 && menuDisplay === false) {
+      handleMenu('resize');
+    } else if (widthScreen <= 800 && menuDisplay === true) {
+      handleMenu('resize');
+    }
+  }, [widthScreen]);
```

Comment:
This whole code is impressive (and it works !), but it's clearly overkill.
- People seldom resize their browser width
- Your `useEffect` would be executed for every change of said width -> several hundreds of times for a mouse manipulation :fearful: 
- We already saw a tool about this (media queries FTW !)